src/canard.c

```diff
@@ -1,7 +1,919 @@
+        rxstate->calculated_crc = crcAdd(rxstate->calculated_crc, frame->data, frame->data_len - 1);
+    }
+    else
+    {
+        CanardRxTransfer rxtransfer = {
+            .timestamp_usec = timestamp_usec,
+            .payload_head = rxstate->buffer_head,
+            .payload_middle = rxstate->buffer_blocks,
+            .payload_tail = frame->data,
+            .middle_len = rxstate->payload_len - CANARD_RX_PAYLOAD_HEAD_SIZE,
+            .payload_len = rxstate->payload_len + frame->data_len - 1,
+            .data_type_id = data_type_id,
+            .transfer_type = transfer_type,
+            .priority = priority,
+            .source_node_id = source_node_id
+        };
+        rxstate->calculated_crc = crcAdd(rxstate->calculated_crc, frame->data, frame->data_len - 1);
+        // crc validation
+        if (rxstate->calculated_crc == rxstate->payload_crc)
+        {
+            ins->on_reception(ins, &rxtransfer);
+        }
+        else
+        {
+            canardReleaseRxTransferPayload(ins, &rxtransfer);
+        }
+        canardPrepareForNextTransfer(rxstate);
+        return;
+    }
+    rxstate->next_toggle ^= 1;
+}
+
+/**
+ * Traverses the list of transfers and removes those that were last updated more than
+ * timeout_usec microseconds ago
+ */
+void canardCleanupStaleTransfers(CanardInstance* ins, uint64_t current_time_usec)
+{
+    CanardRxState* prev = ins->rx_states, * state = ins->rx_states;
+    while (state != NULL)
+    {
+        if ((current_time_usec - state->timestamp_usec)>TRANSFER_TIMEOUT_USEC) // two seconds
+        {
+            if (state==ins->rx_states)
+            {
+                canardReleaseStatePayload(ins, state);
+                ins->rx_states = ins->rx_states->next;
+                canardFreeBlock(&ins->allocator, state);
+                state = ins->rx_states;
+                prev = state;
+            }
+            else
+            {
+                canardReleaseStatePayload(ins, state);
+                prev->next = state->next;
+                canardFreeBlock(&ins->allocator, state);
+                state = prev->next;
+            }
+            continue;
+        }
+        prev = state;
+        state = state->next;
+    }
+}
+
+/**
+ * Reads bits (1 to 64) from RX transfer buffer.
+ * This function can be used to decode received transfers.
+ * Note that this function does not need the library's instance object.
+ */
+uint64_t canardReadRxTransferPayload(const CanardRxTransfer* transfer,
+                                     uint16_t bit_offset,
+                                     uint8_t bit_length)
+{
+    uint64_t bits = 0;
+    int shift_val =  bit_length - 8;
+    if (transfer->payload_len > 7)      //multi frame
+    {
+        CanardBufferBlock* block = transfer->payload_middle;
+        int i;
+        uint8_t index = 0;
+
+        if (CANARD_RX_PAYLOAD_HEAD_SIZE > 0)    //head
```

Comment:
This condition will always be true by design. Is the check intentional, or is it a mistake?